CppExport/src/dependency_analysis/DependencyAnalyzer.cpp

```diff
@@ -89,4 +89,43 @@ QList<DependencyComposite*> DependencyAnalyzer::mergeUnits(QList<DependencyUnit*
 	return nameToCompositeMap.values();
 }
 
+template <class T>
+QList<T*> DependencyAnalyzer::topologicalSort(QHash<T*, QSet<T*>> graph)
+{
+	QHash<T*, QSet<T*>> reverseGraph;
+	for (auto it = graph.begin(); it != graph.end(); it++)
+		for (auto dependency : it.value())
+			reverseGraph[dependency].insert(it.key());
+
+	QList<T*> toBeProcessed;
+	for (auto it = graph.begin(); it != graph.end(); it++)
+		if (!reverseGraph.contains(it.key()))
+			toBeProcessed.append(it.key());
+
+	QList<T*> result;
+	while (!toBeProcessed.empty())
+	{
+		auto n = toBeProcessed.takeFirst();
+		result.append(n);
+
+		for (auto m : graph.value(n))
```

Comment:
Did you test this loop. I feel like the code here is broken and doesn't do topological sort. Having better names might expose the issue. But I feel like here you need to first use the reverseGraph and then the graph (as opposed to the other way around which is the current version).